### PR TITLE
Bring back standard read filters to Mutect2

### PR DESCRIPTION
**Pull request by takutosato**
We apply these filters in GATK3. Also moved `HaplotyeeCallerEngine.makeStandardHCReadFilters` to AssemblyBasedCallerUtils.java.

### CODE REVIEW

**Comment by takutosato** [comment]
@davidbenjamin could you review?

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java [hunk 1]
```diff
@@ -269,4 +273,27 @@ public static AssemblyResultSet assembleReads(final AssemblyRegion region,
             throw e;
         }
     }
+
+    /**
+     * @return the default set of read filters for HC (includes the MQ threshold filter) or M2 (does not include the MQ filter)
+     */
+    public static List<ReadFilter> makeStandardReadFilterList(final boolean includeMQThresholdFilter) {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java]: I'd rather not have the make read filters methods for Mutect and HaplotypeCaller entangled like this -- it is very problematic for adding additional filters in the future to one tool but not the other. Can you restore `HaplotyeeCallerEngine.makeStandardHCReadFilters()`, and give Mutect a completely separate method for creating its filters?

Comment (takutosato) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java]: @droazen will do. Thanks for reviewing the changes. @davidbenjamin please wait for my next commit.

**Comment by takutosato** [comment]
@davidbenjamin this PR is ready for review again.

**Comment by davidbenjamin** [comment]
@takutosato looks good.

**Pull request merged**